### ISSUE TITLE
Host header position causes certain sites to not respond

### ISSUE THREAD

**Issue by Hemable1** [aio-libs/aiohttp#3265]
Outbound requests add the `Host` header last instead of first which causes an issue fetching certain sites. Normally this shouldn't matter, however I'm coming across servers that won't respond unless it's defined first as it is in a browser.

Example:
This is through a browser and successfully responds:
```
GET / HTTP/1.1
Host: www.accuweather.com
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:61.0) Gecko/20100101 Firefox/61.0
Accept: */*
Accept-Encoding: gzip, deflate
Connection: close

```

This is through aiohttp and **does not** respond (notice the Host header position):
```
GET / HTTP/1.1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:61.0) Gecko/20100101 Firefox/61.0
Accept: */*
Accept-Encoding: gzip, deflate
Host: www.accuweather.com
Connection: close

```

To replicate this behavior use the the following code and notice it will time out.

```
import asyncio
import aiohttp

async def req(url):
    headers = {
        'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:61.0) Gecko/20100101 Firefox/61.0'
    }
    timeout = aiohttp.ClientTimeout(total=10)

    async with aiohttp.ClientSession(timeout=timeout, headers=headers) as session:
        async with session.get(url, ssl=False) as resp:
            print(resp.status)
            print(await resp.text())

loop = asyncio.get_event_loop()
loop.run_until_complete(req("https://www.accuweather.com"))
```

Tested on:
Windows 7x64
Python 3.7.0
aiohttp 3.3.2

**Comment by asvetlov** [comment]
I'm sorry but the server is not correct.

We can change an order of request headers construction to put Host first but in general there is no guarantee to satisfy other weird servers requirements.

**Comment by Hemable1** [comment]
Yes the server is incorrect, however this should be developed to replicate a browser as closely as possible. There are a few of sites that fall into this category, `https://www.nasdaq.com` is another one. No browsers I've tried had any issues visiting these sites.

**Comment by webknjaz** [comment]
:open_mouth:

**Comment by GenericStudent** [comment]
@asvetlov, I've created a PR with initial attempt at this. I have moved the `Host` header adding under `update_auto_headers` to be before the default headers. This works and I have added a test for it. But I don't feel the test is optimal.

How do I check / test the view / state of the headers as seen from the **server** perspective?

Also I'm not sure what you meant by the third point.

**Comment by asvetlov** [comment]
Regarding tests: please take a look on https://github.com/aio-libs/aiohttp/blob/master/tests/test_client_functional.py
On server side `request.headers` has request's headers in the same order as they are sent.

The third point is about `client = aiohttp.ClientSession({'host': 'overridden.host.com'})`. It is weird, please don't care about

**Comment by asvetlov** [comment]
Fixed by #3342
